src/libproc_macro/lib.rs

```diff
@@ -200,6 +201,72 @@ impl Span {
     pub fn call_site() -> Span {
         ::__internal::with_sess(|(_, mark)| Span(mark.expn_info().unwrap().call_site))
     }
+
+    /// The original source file into which this span points.
+    #[unstable(feature = "proc_macro", issue = "38356")]
+    pub fn file(&self) -> SourceFile {
```

Comment:
Hmm, I need to check the values for the other methods that `DUMMY_SP` might return to make sure they're not nonsense or else pointing into something unrelated to the current macro expansion.